### ISSUE TITLE
Select /etc/os-release ID, VARIANT_ID, and ID_LIKE

### ISSUE THREAD

**Issue by bgilbert** [coreos/fedora-coreos-tracker#21]
`ID=coreos` implies Container Linux, so we shouldn't use that.  `ID=fedora` + `VARIANT_ID=coreos` is defensible, but tools that see `ID=fedora` might expect things like `dnf` or `/etc/sysconfig/network-scripts` to work.  `ID=fedora-coreos` seems harder to justify.  Other options may exist.

Ultimately this seems like a broader Fedora policy question, and maybe it should get some attention at that level.

**Comment by dustymabe** [comment]
cc @mattdm

**Comment by dustymabe** [comment]
`ID=fedora` + `VARIANT_ID=coreos` would be my vote.

**Comment by cgwalters** [comment]
This is going to be a super bikeshed discussion without a bit more data.  I think "tools that see ID=fedora" is a very valid concern.  However, I think a general best practice is to use *feature* detection and not *OS* detection.  For example, people should be using `if test -x /usr/bin/dnf`, not looking for `fedora` in `/usr/lib/os-release` - among other reasons, doing the latter totally breaks with derivatives in play.  (Yes, there's `ID_LIKE` but still)

So it'd be interesting to see if we can think of any concrete examples of tools that actually look for `ID=fedora` (or `ID_LIKE=fedora` !)

To turn this whole discussion around - what would break if we did `ID=$(uuidgen)` for each new build?  What tools are really looking at that field today?

**Comment by mscherer** [comment]
While feature detection is surely more robust, that's also much more complex to handle in various tools that do look at that. For example, ansible do have package manager detection, but on atomic, that's a mess since there is several package manager (atomic, docker, rpm-ostree and flatpak, for silverblue). 

Having a ID permit at least to make high level choice. I think the ID should be considered as part of a API, and it should mean something (like, when you see that, and that version, then you can expect this). No one benefit from having complex detection logic duplicated in each and every tools that could have to deduce something based on the name.

**Comment by cgwalters** [comment]
> While feature detection is surely more robust, that's also much more complex to handle in various tools that do look at that. For example, ansible do have package manager detection, 

Note that one can also use flatpak on e.g. [classic Fedora](https://pagure.io/Fedora-Council/tickets/issue/208) - not to mention Ubuntu, etc.   So such ID-based detection is already broken.

**Comment by dustymabe** [comment]
currently this is implemented as `ID=fedora` + `VARIANT_ID=coreos` in the [fedora-coreos-release rpm](https://src.fedoraproject.org/rpms/fedora-release/blob/40797cacf6fe34e60f6738036cee77f3b5892e7b/f/fedora-release.spec#_274-279):

```
[core@coreos ~]$ cat /etc/os-release | grep -Pi '(variant|ID)'
ID=fedora
VERSION_ID=29
PLATFORM_ID="platform:f29"
VARIANT="CoreOS"
VARIANT_ID=coreos

```

PR to update the design doc: https://github.com/coreos/fedora-coreos-tracker/pull/46

**Comment by dustymabe** [comment]
closed in #46
